android/runtime/v8/src/native/common.mk

```diff
@@ -33,6 +33,32 @@ CLEAN_OBJ := if exist $(OBJ_DIR) (rd /s /q $(subst /,\\,$(OBJ_DIR)) && mkdir $(s
 endif
 
 LDLIBS := -L$(SYSROOT)/usr/lib -ldl -llog -L$(TARGET_OUT)
+
+# optimize output size
+CFLAGS += -Os -flto -ffunction-sections -fdata-sections
+LDLIBS += -Wl,--gc-sections,--strip-all
+
+# exclude v8 libraries
+LDLIBS += -Wl,--exclude-libs=libv8_libbase
```

Comment:
So let me first preface this by saying I know nothing about C/C++ flags, linker flags, etc.

Would a change like this affect the ability to build native modules?

Also to note: in the 8.0 timeframe I'm moving to a newer V8 building that generates a single static library (`v8_monolith.a`)

Reply:
I tested this with a native module and everything seems to function fine. It's the symbols from `libv8_base` (commented out) which are required by native modules.

I guess we will need to take another look at this once we move to a single library. We may not be able to optimize it as much?